go-controller/hybrid-overlay/pkg/controller/honode_linux.go

```diff
@@ -0,0 +1,124 @@
+package controller
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/types"
+	houtil "github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/util"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"github.com/vishvananda/netlink"
+	kapi "k8s.io/api/core/v1"
+	listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+)
+
+// HONodeController is the node hybrid overlay controller
+type HONodeController struct {
+	kube            kube.Interface
+	machineID       string
+	nodeName        string
+	localNodeCIDR   *net.IPNet
+	localNodeIP     net.IP
+	remoteSubnetMap map[string]string // Maps a remote node to its remote subnet
+}
+
+// newHONodeController returns a node handler that listens for node events
+// so that Add/Update/Delete events are appropriately handled.
+func newHONodeController(kube kube.Interface,
+	nodeName string,
+	nodeLister listers.NodeLister,
+	podLister listers.PodLister,
+) (nodeController, error) {
+
+	node, err := kube.GetNode(nodeName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &HONodeController{
+		kube:            kube,
+		machineID:       node.Status.NodeInfo.MachineID,
+		nodeName:        node.Name,
+		remoteSubnetMap: make(map[string]string),
+	}, nil
+}
+
+// AddNode set annotations about its VTEP and gateway MAC address to its own node object
+func (n *HONodeController) AddNode(node *kapi.Node) error {
+	if node.Name == n.nodeName {
+		cidr, nodeIP := getNodeSubnetAndIP(node)
+		if (cidr != nil && !houtil.SameIPNet(cidr, n.localNodeCIDR)) || (nodeIP != nil && nodeIP.Equal(n.localNodeIP)) {
```

Comment:
same here

Reply:
I would add a comment explaining why we only act when we notice a change in CIDR or IP